geode-core/src/test/java/org/apache/geode/cache/client/internal/CacheServerSSLConnectionDUnitTest.java

```diff
@@ -178,11 +180,20 @@ public void setUpClientVM(String host, int port, boolean cacheServerSslenabled,
     gemFireProps.put(SERVER_SSL_CIPHERS, cacheServerSslciphers);
     gemFireProps.put(SERVER_SSL_REQUIRE_AUTHENTICATION, String.valueOf(cacheServerSslRequireAuth));
 
-    gemFireProps.put(SERVER_SSL_KEYSTORE_TYPE, "jks");
-    gemFireProps.put(SERVER_SSL_KEYSTORE, keyStorePath);
-    gemFireProps.put(SERVER_SSL_KEYSTORE_PASSWORD, "password");
-    gemFireProps.put(SERVER_SSL_TRUSTSTORE, trustStorePath);
-    gemFireProps.put(SERVER_SSL_TRUSTSTORE_PASSWORD, "password");
+
+    if (clientHasTrustedKeystore) {
+      gemFireProps.put(SERVER_SSL_KEYSTORE_TYPE, "jks");
+      gemFireProps.put(SERVER_SSL_KEYSTORE, keyStorePath);
+      gemFireProps.put(SERVER_SSL_KEYSTORE_PASSWORD, "password");
+      gemFireProps.put(SERVER_SSL_TRUSTSTORE, trustStorePath);
+      gemFireProps.put(SERVER_SSL_TRUSTSTORE_PASSWORD, "password");
+    } else {
+      gemFireProps.put(SERVER_SSL_KEYSTORE_TYPE, "jks");
+      gemFireProps.put(SERVER_SSL_KEYSTORE, "");
```

Comment:
Is `""` a singleton value recognized as a special case? Or does it work because it's an invalid path?